templates/form-fields/file-field.php

```diff
@@ -4,7 +4,7 @@
 $field_name         = isset( $field['name'] ) ? $field['name'] : $key;
 $field_name         .= ! empty( $field['multiple'] ) ? '[]' : '';
 
-if ( ! empty( $field['ajax'] ) ) {
+if ( ! empty( $field['ajax'] ) && job_manager_user_can_upload_file_via_ajax() ) {
```

Comment:
So my solution was to just add this function and call it from my template overrides:

```php
if( ! function_exists( 'field_editor_user_can_upload_file_via_ajax' ) ){

	/**
	 * Checks if the user can upload a file via the Ajax endpoint.
	 *
	 * This is a function added for compatibility with older versions of WP Job Manager, as this function
	 * has to be called in the file-field.php template file to prevent unauthorized ajax uploads.
	 *
	 * @since @@since
	 * @return bool
	 */
	function field_editor_user_can_upload_file_via_ajax() {

		// Added in core @since 1.26.2
		if( function_exists( 'job_manager_user_can_upload_file_via_ajax' ) ){
			return job_manager_user_can_upload_file_via_ajax();
		}

		$can_upload = is_user_logged_in() && job_manager_user_can_post_job();

		/**
		 * Override ability of a user to upload a file via Ajax.
		 *
		 * @since @@since
		 *
		 * @param bool $can_upload True if they can upload files from Ajax endpoint.
		 */
		return apply_filters( 'job_manager_user_can_upload_file_via_ajax', $can_upload );
	}

}
```